clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -35,16 +36,17 @@ private object ApiClient {
 
   /** @return an ApiClient, reusing an existing one for this URL if possible.
    */
-  def get(
-      apiUrl: String,
-      accessKey: String,
-      secretKey: String,
-      connectionTimeoutSec: String = "",
-      readTimeoutSec: String = ""
-  ): ApiClient = clients.get(
-    ClientKey(apiUrl, accessKey),
+  def get(conf: APIConfigurations): ApiClient = clients.get(
+    ClientKey(conf.apiUrl, conf.accessKey),
     new Callable[ApiClient] {
-      def call() = new ApiClient(apiUrl, accessKey, secretKey, connectionTimeoutSec, readTimeoutSec)
+      def call() = new ApiClient(
+        new APIConfigurations(conf.apiUrl,
+                              conf.accessKey,
+                              conf.secretKey,
+                              conf.connectionTimeoutSec,
+                              conf.readTimeoutSec
+                             )
```

Comment:
Isn't this just a copy of APIConfigurations?

Reply:
More details: all members of `conf` are immutable, it is a value class.  So there is no value (😀) in copying it.

Reply:
Changed APIConfigurations into a case class

> Thanks!
> 
> As we discussed off-review yesterday, I would _much_ rather that we did either one of these:
> 
> * Use a proper duration parser, so I can configure a timeout to be something like `1m10s`.  `java.time.Duration.parse("PT" + s)` will probably do this.
> * Add a proper duration parser later, and rename the _currently used_ configuration options to end `_secs` or `seconds`, to make configuration clear.  The variables that carry these unitless numbers should also have units in their names.  Getting the wrong value is too easy.
> 
> About default timeout values: assuming we always get these from a Hadoop configuration. So we might use [`org.hadoop.Configuration.get(name String, defaultValue String)`](https://hadoop.apache.org/docs/r2.7.1/api/org/apache/hadoop/conf/Configuration.html#get(java.lang.String,%20java.lang.String)) to get-with-default, and get rid of the entire setting logic. WDYT?
> 
> I prefer to keep

Thanks! 

I changed the property name as suggested in option number 2. 
as for the default timeout values, since these are defaults of the API client, I'm avoiding hard-coding it somewhere else. that is, we are not getting them from hadoop configurations by default, makes sense?